src/backend/chat/commands/command-manager.ts

```diff
@@ -0,0 +1,522 @@
+
+        const commandDb = this.getCommandsDb();
+
+        try {
+            commandDb.push(`/customCommands/${command.id}`, command);
+        } catch (err) {}
+
+        const existingCommandIndex = this._commandCache.customCommands.findIndex(c => c.id === command.id);
+        if (existingCommandIndex === -1) {
+            this._commandCache.customCommands.push(command);
+        } else {
+            this._commandCache.customCommands[existingCommandIndex] = command;
+        }
+
+        frontendCommunicator.send("custom-command-saved", command);
+    }
+
+    /**
+     * Imports a custom command, like from a setup or a previous version of Firebot
+     *
+     * @param command The `CommandDefinition` for the custom command to import
+     */
+    saveImportedCustomCommand(command: CommandDefinition): void {
+        logger.debug(`Saving imported command: ${command.trigger}`);
+
+        if (command.id == null || command.id === "") {
+            command.createdBy = "Imported";
+        } else {
+            command.lastEditBy = "Imported";
+        }
+
+        this.saveCustomCommand(command, "Imported");
+    }
+
+    /**
+     * Saves all the specified custom commands
+     *
+     * @param commands An array of `CommandDefinition` objects to save
+     */
+    saveAllCustomCommands(commands: CommandDefinition[]): void {
+        try {
+            const commandDb = this.getCommandsDb();
+            const customCommandsObj = commands.reduce((acc, command) => {
+                acc[command.id] = command;
+                return acc;
+            }, {});
+            commandDb.push("/customCommands", customCommandsObj);
+        } catch (err) {}
+
+        this._commandCache.customCommands = commands;
+
+        frontendCommunicator.send("custom-commands-updated");
+    }
+
+    /**
+     * Deletes a custom command
+     *
+     * @param id ID of the custom command to delete
+     */
+    deleteCustomCommand(id: string): void {
+        const commandDb = this.getCommandsDb();
+
+        if (id == null) {
+            return;
+        }
+
+        try {
+            commandDb.delete(`/customCommands/${id}`);
+        } catch (err) {
+            logger.warn("error when deleting command", err.message);
+        }
+
+        this._commandCache.customCommands = this._commandCache.customCommands.filter(c => c.id !== id);
+
+        frontendCommunicator.send("custom-command-deleted", id);
+    }
+
+    /**
+     * Deletes a custom command by its trigger
+     *
+     * @param trigger The trigger of the custom command to delete
+     */
+    removeCustomCommandByTrigger(trigger: string): void {
+        const command = this._commandCache.customCommands.find(c => c.trigger === trigger);
+        if (command != null) {
+            this.deleteCustomCommand(command.id);
+        }
+    }
+
+    /**
+     * Refreshes the command cache
+     *
+     * @param retry The retry count. Default is `1`.
+     */
+    refreshCommandCache(retry = 1): void {
```

Comment:
I'm pretty sure we can strip out the weird retry logic from this function now that we are read/writing data in the backend only. This was a hacky-ish solution to deal with weirdness steaming from the renderer and main services both trying to access the same file and we don't do it anywhere else so we shouldn't need it here.